### DISCUSSION TITLE
Reach study failed to load custom mesh generated from Solidworks

### DISCUSSION THREAD

**Discussion by tiagostmec** [ros-industrial/reach_ros#13] [Q&A] [answered]
Hi, im trying to use reach_ros on a custom .ply and .pcd generated already by the readme command.
Im using the demo robot, and the demo config for the test. if i use the given .ply from the gif readme example works fine. 
But when i try to generate a new worktable using the same robot and just trying to generate a new ply from a CAD,  
the error code rises  on  roslaunch reach_ros start.launch config_file:=<config_file.yaml> config_name:=<arbitrary_config>

Any help from here? Im messing with the pcl lib?

**Comment by tiagostmec** [comment]
I got the problem, if someone is using solidworks, take care because when you import your cad as a .ply object sldw mess up the measures and we cant load it on rviz.
For note i was using solidworks 2019 for CAD
and ubuntu 20.04 for reach_ros.

**Reply by marip8** [reply, marked as answer]
Was your .ply file exported with the wrong scale, or was it unable to be loaded in the reach study? I have seen strange issues in the past with .ply files generated by Solidworks on Linux machines. My fix in the past has been to open them in a Linux viewer (e.g., Meshlab) and re-save them.

**Comment by tiagostmec** [comment]
> Was your .ply file exported with the wrong scale, or was it unable to be loaded in the reach study? 

I got both, wrong scale in solid works happens because it uses mm as base, and when imported as a mesh it says 200mm = 200m, easy to  rescale. But, the error -11 happened because my ply import from solid works was messed on measures.


 > I have seen strange issues in the past with .ply files generated by Solidworks on Linux machines. My fix in the past has been to open them in a Linux viewer (e.g., Meshlab) and re-save them.

And yes ! thats the solution, solid works mess up when you save .ply directly, then i opened it on a mesh lab and re-saved it as ply. The problem is on solidworks side probally. :D
